packages/checkout/registry/index.js

```diff
@@ -53,45 +52,13 @@ const getCheckoutFilters = ( filterName ) => {
 export const __experimentalApplyCheckoutFilter = ( {
```

Comment:
Given that this function now contains a hook, I think we should rename it to `useExperimentalApplyCheckoutFilter` or something similar. React + ESLint apply some extra-checks to functions whose name starts with `use` so they follow the [rules of hooks](https://reactjs.org/docs/hooks-rules.html). The name will need to be updated in the [docs](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/blocks/feature-flags-and-experimental-interfaces.md) as well.

Reply:
It seems the rule of hooks ignores functions that start with an underscore so we should be good?
I'd prefer we leave it as is for now, I didn't spot any issues with it.